### PR TITLE
Try testing oldest dependency versions in a separate configuration.

### PR DESCRIPTION
**Pull request by bdice**
## Description
In some Slack discussions, we decided it might be good to test newest and oldest dependency versions in separate configurations. This provides a configuration for the "oldest" tests.

Along the way, I found quite a few places where our stated package requirements were inaccurate. 😬

This PR separates our tests into "newest" (most of the configurations) and "oldest," which is the lowest version numbers that we know we support. I recommend the following policies for testing in the future:

The oldest dependencies should only be changed in a limited number of cases:
- Dropping Python versions (only the earliest supported version needs to be tested with "oldest" dependencies)
- Adding new features that rely on newer dependency versions

For the newest dependencies:
- New test configurations should be added when new Python versions are released and added to the list of supported versions
- Pinned testing dependencies are automatically updated by dependabot
- Configurations can be dropped from CI if "newest" dependencies drop support. I am hoping that our dependencies drop support no faster than NEP 29 would. I am also hoping we only need to support releasing for configurations that are covered by NEP 29 at release time.

## How Has This Been Tested?
Added new CI configuration.

## Types of changes

- [x] New feature (non-breaking change which adds or improves functionality)

## Checklist:
- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -27,15 +27,39 @@ references:
...
-        # won't be dependent on them updating wheels for specific new Python
-        # versions.
-        ${PYTHON} -m pip install --user --progress-bar off scikit-build==0.10.0
-        ${PYTHON} -m pip install --user --progress-bar off -U -r requirements-testing.txt
+        # Create a virtual environment to ignore the CI image's pre-installed packages
+        ${PYTHON} -m pip install --progress-bar off --upgrade virtualenv --user
+        mkdir -p ./venv
+        ${PYTHON} -m virtualenv ./venv --clear
+        . venv/bin/activate
+        ${PYTHON} -m pip install --progress-bar off --upgrade pip virtualenv
```
Comment (b-butler) [inline .circleci/config.yml]: Why are we upgrading `virtualenv` again?

**Review by b-butler — COMMENTED** [review]
I only have one comment otherwise the PR LGTM.

File: .circleci/config.yml (hunk 1 above)
Comment (bdice) [reply .circleci/config.yml]: That might have been an oversight on my part. I'll change it to only upgrade pip.
```suggestion
        ${PYTHON} -m pip install --progress-bar off --upgrade pip
```

**Comment by bdice** [comment]
I fixed the one comment from @b-butler and I will merge this when tests pass.

**Pull request merged**